### ISSUE TITLE
[FrameworkBundle] Config diff between 2.8 and 3.0.

### ISSUE THREAD

**Issue by ogizanagi** [symfony/symfony#16570]
Just to mention [the issue we encountered while updating EasyAdminBundle for 3.0 support](https://github.com/javiereguiluz/EasyAdminBundle/pull/566#issuecomment-155535288), keeping the exact same configuration (in tests):

```yml
framework:
    secret:          secret
    translator:      ~
    default_locale:  "%locale%"
    test:            ~
    router:          { resource: "%kernel.root_dir%/config/routing_base.yml" }
    form:            true
    validation:      { enable_annotations: true }
    templating:      { engines: ['twig'] }
    profiler:
        collect: true
    session:
        storage_id: session.storage.mock_file
```

This is the diff between the 2.8 and 3.0 processed configuration:

```diff
< array:23 [
---
> array:21 [
22,23c22,23
<       "enabled" => null
<       "field_name" => null
---
>       "enabled" => false
>       "field_name" => "_token"
47c47
<   "profiler" => array:8 [
---
>   "profiler" => array:5 [
53,55d52
<     "username" => ""
<     "password" => ""
<     "lifetime" => 86400
69,72d65
<   "csrf_protection" => array:2 [
<     "enabled" => false
<     "field_name" => "_token"
<   ]
99,105d91
<   "assets" => array:5 [
<     "version" => null
<     "version_format" => "%%s?%%s"
<     "base_path" => ""
<     "base_urls" => []
<     "packages" => []
<   ]
111c97
```

The issue we encountered was about the `asset` twig function, which isn't loaded in 3.0, because the `framework.assets` key doesn't have the "default values" it used to anymore, due to the framework bundle's [Configuration class](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L48-L122) (the highlighted part is removed in 3.0).

I don't think it was mentioned anywhere. What do you think should be done about that ?

**Comment by csarrazi** [comment]
It seems 3.0 is not the only version affected. See https://github.com/csarrazi/CsaGuzzleBundle/issues/85

**Comment by xabbuh** [comment]
@csarrazi If I understand csarrazi/CsGuzzleBundle#85 correctly, this was an issue caused by the user's code. Is that right?

**Comment by xabbuh** [comment]
@fabpot Assumed this is only different between Symfony 2.8 and 3.0 should we treat this as an intended BC break and thus document it in the upgrade files or do we have to fix the default config (I would say the latter as this is really unexpected for users as we never deprecated the old behaviour if I am not mistaken)?

**Comment by fabpot** [comment]
Assets are disabled by default in 3.0, right? We can reenable it by default indeed to avoid such problems.

**Comment by csarrazi** [comment]
Actually, it is a BC break from Symfony 2.7 to 2.8. By default, assets are disabled on both Symfony 2.8 and 3.0, which means one needs to add the `framework.assets: ~` key in the configuration, for most bundles to work (if using the `asset` helper, which is the case most of the time).

**Comment by xabbuh** [comment]
I cannot reproduce the wrong behaviour on Symfony 2.8 (and the code you linked to is even from Symfony 2.8). For 3.0 and higher I opened a pull request to fix this (see #17506).
